### DISCUSSION TITLE
Strongswan 5.9 not using configured cipher proposals

### DISCUSSION THREAD

**Discussion by felartu** [strongswan/strongswan#2266] [Q&A] [answered]
Hi,

Rocky Linux 9
Strongswan 5.9.10

I am seeing this strange behavior in Strongswan where I configure some ESP proposals and they are not being picked up by Strongswan during the handshake.

msg = received proposals: ESP:AES_CBC_256/HMAC_SHA1_96/MODP_1024/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA1_96/MODP_1024/
NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA1_96/MODP_1024/NO_EXT_SEQ

msg = configured proposals: ESP:AES_CBC_256/HMAC_SHA1_96/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_128/AES_CBC_192/AES_CBC_2
56/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/HMAC_SHA1_96/AES_XCBC_96/NO_EXT_SEQ, ESP:AES_GCM_16_128/AES_G
CM_16_192/AES_GCM_16_256/NO_EXT_SEQ

 msg = no acceptable proposal found

in my swanctl.conf I have this:

esp_proposals = aes128-aes256-aes192-sha-sha256-modp1024-modp2048,aes128-sha-modp1024,aes256-sha-modp1024,aes256-sha-modp2048,aes256-sha-modp1024,default

The ciphers show up in swanctl --list-algs output.

Yet still Strongswan fails to use these ciphers, what am I missing here?

Note: This is a 3500+ tunnels strongswan instance

Logs attached.
[logs.txt](https://github.com/strongswan/strongswan/files/15498965/logs.txt)
[charon_debug.txt](https://github.com/strongswan/strongswan/files/15499363/charon_debug.txt)

**Comment by tobiasbrunner** [comment, marked as answer]
Please update your post and add the complete config and logs (see #196).

**Reply by felartu** [reply]
Logs attached

**Reply by tobiasbrunner** [reply]
That log is not what I asked for (see the config snippet in #196 and post the complete log).

**Reply by felartu** [reply]
I added charon debug logs from code snippet.

**Reply by tobiasbrunner** [reply]
Still not what I asked for. I need the complete log, from starting the daemon (and loading the config) to the error.

**Reply by felartu** [reply]
That will require some downtime for 3500 sites, let me schedule that and come back with starting daemon logs until the error happens.
Expect a very long log since all sites will try to reconnect simultaneously.

**Reply by tobiasbrunner** [reply]
Please read the log yourself then first. Because you probably made some simple mistake that you could easily fix yourself if you read the log (or maybe you didn't even reload the config or restart the affected SA).

**Reply by felartu** [reply]
Is there any difference regarding proposal changes with `swanctl --load-all` vs `systemctl restart strongswan` ?

**Reply by tobiasbrunner** [reply]
Depends on what you changed and whether there are active SAs that use a modified config. The config reload by the first command won't affect existing SAs even when rekeying (except if only the child-reated settings have changed, but if you also changed e.g. the IKE proposal, you have to restart the IKE_SA for it to fully take effect).

**Reply by felartu** [reply]
I restarted the daemon and the correct proposals were picked up and the issue is fixed now, thanks.
